## Read, check and changepixel

### Read image first

In [1]:
import cv2
path = "I:\\openCV\\test.jpg"
# be careful of path and \\
image = cv2.imread(path,flags=0)
# check if the image load successfully or not 
if image is None:
    print("Fail to load")
else:
    cv2.imshow('image', image)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

### Check the grayscale value

In [21]:
p = image[88,152] #88 row 152 column
p

172

Return value is 172. The grayscale usually start from 0 to 255, 0 means black, and 255 means white.

### Check BGR value.
#### read image first

In [3]:
image1 = cv2.imread(path) # no flag means BGR image
# check if the image load successfully or not 
if image1 is None:
    print("Fail to load")
else:
    cv2.imshow('image', image1)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

In [4]:
blue = image1[78,125,0] # the 3rd value is color tune
blue

158

In [8]:
green = image1[78,125,1]
green

169

In [9]:
red = image1[78,125,2]
red

167

In [10]:
value = image1[78,125]
value # the return value is 3 color value, order by BGR

array([158, 169, 167], dtype=uint8)

## Change pixel

### Grayscale pixel change

In [15]:
image[88,99] = 255
cv2.imshow('image', image)
cv2.waitKey(0)
cv2.destroyAllWindows()

### BGR pixel change

#### change one by one

In [12]:
image1[88,99,0]=255

In [13]:
image1[88,99,1]=255

In [14]:
image1[88,99,2]=255

#### or change at the same time

In [17]:
image1[88,100]=[255,255,255]
cv2.imshow('image', image1)
cv2.waitKey(0)
cv2.destroyAllWindows()

#### Try to print the value of [88,100],[88,99]

In [18]:
image1[88,100]

array([255, 255, 255], dtype=uint8)

In [19]:
image1[88,99]

array([255, 255, 255], dtype=uint8)

#### We can also change area by using list slice

In [20]:
image1[100:150,100:150]=[255,255,255]
cv2.imshow('sliced', image1)
cv2.waitKey(0)
cv2.destroyAllWindows()

We can see a white area in the image, that is the area we changed